components/script/script_thread.rs

```diff
@@ -999,23 +1157,30 @@ impl ScriptThread {
                     // step 7.8
                     if !animation_ticks.contains(&pipeline_id) {
                         animation_ticks.insert(pipeline_id);
+                        let t = Task::new(None, Some(pipeline_id));
+                        self.task_times.borrow_mut().push(t);
                         sequential.push(event);
                     }
                 }
                 FromConstellation(ConstellationControlMsg::SendEvent(
                         _,
                         MouseMoveEvent(_))) => {
+                    let t = Task::new(None, self.message_to_pipeline(&event));
                     match mouse_move_event_index {
                         None => {
                             mouse_move_event_index = Some(sequential.len());
+                            self.task_times.borrow_mut().push(t);
                             sequential.push(event);
                         }
                         Some(index) => {
+                            self.task_times.borrow_mut().push(t);
                             sequential[index] = event
                         }
                     }
                 }
                 _ => {
+                    let t = Task::new(None, self.message_to_pipeline(&event));
+                    self.task_times.borrow_mut().push(t);
```

Comment:
It's not clear to me why we want to set the start time for these events at this point in time. If we push two events named A and B onto sequential, then run event A followed by event B, doesn't this mean that B's duration will include the time taken to run A?